src/Features/CSharp/Portable/CodeRefactorings/SyncNamespace/CSharpSyncNamespaceService.cs

```diff
@@ -0,0 +1,294 @@
+                {
+                    eofToken = eofToken.WithPrependedLeadingTrivia(
+                        triviaFromNamespaceDecl.openingTrivia.Concat(triviaFromNamespaceDecl.closingTrivia));
+                }
+
+                return compilationUnit.Update(
+                    compilationUnit.Externs.AddRange(namespaceDeclaration.Externs), 
+                    compilationUnit.Usings.AddRange(namespaceDeclaration.Usings), 
+                    compilationUnit.AttributeLists, 
+                    members,
+                    eofToken).WithAdditionalAnnotations(Formatter.Annotation);
+            }
+
+            // Change namespace name
+            return root.ReplaceNode(namespaceDeclaration, 
+                namespaceDeclaration.WithName(
+                    CreateNameSyntax(targetNamespaceParts, aliasQualifier: null, targetNamespaceParts.Length - 1)
+                    .WithTriviaFrom(namespaceDeclaration.Name)
+                    .WithAdditionalAnnotations(WarningAnnotation, Formatter.Annotation)));
+        }
+
+        private static string GetAliasQualifierOpt(SyntaxNode name)
+        {
+            while (true)
+            {
+                switch (name.Kind())
+                {
+                    case SyntaxKind.QualifiedName:
+                        name = ((QualifiedNameSyntax)name).Left;
+                        continue;
+                    case SyntaxKind.AliasQualifiedName:
+                        return ((AliasQualifiedNameSyntax)name).Alias.Identifier.ValueText;
+                }
+
+                return null;
+            }
+        }
+
+        private NameSyntax CreateNameSyntax(ImmutableArray<string> namespaceParts, string aliasQualifier, int index)
+        {
+            var part = namespaceParts[index].EscapeIdentifier();
+            var namePiece = SyntaxFactory.IdentifierName(part);
+
+            if (index == 0)
+            {
+                return aliasQualifier == null ? (NameSyntax)namePiece : SyntaxFactory.AliasQualifiedName(aliasQualifier, namePiece);
+            }
+            else
+            {
+                return SyntaxFactory.QualifiedName(CreateNameSyntax(namespaceParts, aliasQualifier, index - 1), namePiece);
+            }
+        }
+
+        /// <summary>
+        /// return trivia attached to namespace declaration. 
+        /// Leading trivia of the node and trivia around opening brace, as well as
+        /// trivia around closing brace are concatenated together respectively.
+        /// </summary>
+        private static (List<SyntaxTrivia> openingTrivia, List<SyntaxTrivia> closingTrivia) 
+            GetOpeningAndClosingTriviaOfNamespaceDeclaration(NamespaceDeclarationSyntax namespaceDeclaration)
+        {
+            var openingTrivia = namespaceDeclaration.GetLeadingTrivia().ToList();
+            openingTrivia.AddRange(namespaceDeclaration.OpenBraceToken.LeadingTrivia);
+            openingTrivia.AddRange(namespaceDeclaration.OpenBraceToken.TrailingTrivia);
```

Comment:
both of these were suprising to me.  do we expect those to ever happen?

Reply:
Leading trivia is unlikely, but I think trailing trivia might not be uncommon.

Reply:
that would be:

```c#
namespace A.B.C
{ // trailing trivia
    int Whatever => 0;
}
```

That seems like a fairly strange thing to try to support IMO.  Have you seen this pattern before?

Reply:
It's not common, but I definitely saw people putting comments at the same line as the opening brace.

Reply:
ok! :)  If you've seen this (and there are tests for this case), i'm a-ok with it :)